flintrock/ec2.py

```diff
@@ -57,33 +58,47 @@ def __init__(
             vpc_id: str,
             master_instance: 'boto3.resources.factory.ec2.Instance',
             slave_instances: "List[boto3.resources.factory.ec2.Instance]",
+            use_private_vpc,
```

Comment:
I'm not sure that the `self.vpc_id` is enough to determine if the subnet that the cluster is in auto-assigns public ips. However, I can get the subnet id from the master instance and then query the subnet to get the desired result. What do you think?